sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -610,17 +615,25 @@ public ServiceBusSessionProcessorClientBuilder sessionProcessor() {
     void onClientClose() {
         synchronized (connectionLock) {
             final int numberOfOpenClients = openClients.decrementAndGet();
-            logger.info("Closing a dependent client. # of open clients: {}", numberOfOpenClients);
+            logger.atInfo()
+                .addKeyValue("numberOfOpenClients", numberOfOpenClients)
+                .log("Closing a dependent client.");
 
             if (numberOfOpenClients > 0) {
                 return;
             }
 
             if (numberOfOpenClients < 0) {
-                logger.warning("There should not be less than 0 clients. actual: {}", numberOfOpenClients);
+                logger.atWarning()
+                    .addKeyValue("numberOfOpenClients", numberOfOpenClients)
+                    .log("There should not be less than 0 clients.");
             }
 
-            logger.info("No more open clients, closing shared connection [{}].", sharedConnection);
+            logger.atInfo()
+                // TODO: it doesn't look right to log ServiceBusConnectionProcessor instance. What should we log here instead?
+                .addKeyValue("sharedConnection", sharedConnection)
```

Comment:
logging sharedConnection doesn't look right (it's a complex object without toString overload). What should we do here

Reply:
I think removing this key-value pair is fine. We only need to log that the underlying connection is closed.